zoo/board_games/mcts_bot.py

```diff
@@ -61,52 +77,70 @@ def backpropagate(self, reward):
         pass
 
     def is_fully_expanded(self):
+        """
+        Overview:
+            Check whether the node is fully expanded. Whenever a new child node is selected first time, an element 
+            is popped out from the list of legal actions. When the list becomes empty, it indicates that all child 
+            nodes have been selected. This means that the parent node has been completed expanded.
+        """
         return len(self.legal_actions) == 0
 
     def best_child(self, c_param=1.4):
         '''
         Overview:
-                 - computer ucb score.
-                    ucb = (q / n) + c_param * np.sqrt((2 * np.log(visited_num) / n))
-                    - q: The estimated value of Node. 
-                    - n: The simulation num of Node.
-                    - visited_num: The visited num of Node
-                    - c_param: constant num=1.4.
-                 - Select the node with the highest ucb score.
+            Find the best child node which has the highest ucb score.
+            {UCT}(v_i, v) = \frac{Q(v_i)}{N(v_i)} + c \sqrt{\frac{\log(N(v))}{N(v_i)}}
+                - Q(v_i) is the estimated value of the child node v_i.
+                - N(v_i) is a counter of how many times the child node v_i has been on the backpropagation path.
+                - N(v) is a counter of how many times the parent(current) node v has been on the backpropagation path.
+                - c is a parameter which balance exploration and exploitation.
         '''
+        # Calculate the ucb score for every child node in the list.
         choices_weights = [(c.q / c.n) + c_param * np.sqrt((2 * np.log(self.n) / c.n)) for c in self.children]
         self.best_action = self.parent_action[np.argmax(choices_weights)]
+        # Choose the child node which has the highest ucb score.
         return self.children[np.argmax(choices_weights)]
 
     def rollout_policy(self, possible_actions):
+        # randomly choose an action 
```

Comment:
Overview:
            This method implements the rollout policy for a node during the Monte Carlo Tree Search.
            The rollout policy is used to determine the action taken during the simulation phase of the MCTS. In this case, the policy is to randomly choose an action from the list of possible actions.
        Arguments:
            - possible_actions: A list of all possible actions that can be taken from the current state.
        Return: 
            A randomly chosen action from the list of possible actions.